client/src/components/game/gui/GUI.jsx

```diff
@@ -107,35 +97,49 @@ function GUI() {
                 <PanelButton
                   icon={statsIcon}
                   onClick={() => {
-                      setShownPanel(Panels.Stats);
+                      if (shownPanel === Panels.NONE) setShownPanel(Panels.Stats);
+                      else closePanelCallback();
                   }}
                   tooltipText={`${Utils.getTextDef("Stats tooltip")} ( V )`}
                 />
                 <PanelButton
                   icon={tasksIcon}
                   onClick={() => {
-                      setShownPanel(Panels.Tasks);
+                      if (shownPanel === Panels.NONE) setShownPanel(Panels.Tasks);
+                      else closePanelCallback();
                   }}
                   tooltipText={`${Utils.getTextDef("Tasks tooltip")} ( B )`}
                 />
                 <PanelButton
                   icon={mapIcon}
                   onClick={() => {
-                      setShownPanel(Panels.Map);
+                      if (shownPanel === Panels.NONE) setShownPanel(Panels.Map);
+                      else closePanelCallback();
                   }}
                   tooltipText={`${Utils.getTextDef("Map tooltip")} ( M )`}
                 />
+                <PanelButton
+                  icon={chatIcon}
+                  onClick={() => {
+                      if (shownPanel === Panels.NONE) setShownPanel(Panels.Chat);
+                      else closePanelCallback();
+                  }}
+                  tooltipText={`${Utils.getTextDef("Chat tooltip")} ( ENTER )`}
+                />
             </div>
 
             <div className="top-right-corner-cont gui-zoomable">
                 <PanelButton
                   icon={exitIcon}
                   onClick={() => {
                       if (loggedIn) {
-                          setShownPanel(Panels.Account);
+                          if (shownPanel === Panels.NONE) setShownPanel(Panels.Account);
```

Comment:
This is open/close toggle is missing for settings button below.

Reply:
Wait, I didn't quite understand. Did you mean that I need to revert this changes?

Reply:
This same checks added to these buttons should be added to the settings button too for consistency.

```js
<PanelButton
  icon={settingsIcon}
  onClick={() => {
    setShownPanel(Panels.Settings);
  }}
  tooltipText={Utils.getTextDef("Settings tooltip")}
/>
```

Reply:
I see